# Defining the fetal gene program at single cell resolution in dilated cardiomyopathy. 
[![Publication: Circulation](https://img.shields.io/badge/Published%20in-Circulation-red)](link_to_publication_here)  
[![MCRI](https://img.shields.io/badge/Institute-MCRI-blue)](https://www.mcri.edu.au)  
[![Interactive Portal](https://img.shields.io/badge/Explore%20Data-HeartExplorer.org-brightgreen)](https://www.HeartExplorer.org)  
[![workflowr](https://img.shields.io/badge/Workflowr-Project-ff69b4)](https://github.com/jdblischak/workflowr)

This is a large single nuclei RNA-seq dataset from the Heart Regeneration group at MCRI. The samples were prepared by Evangelyn Sim and the data was processed and sequenced at the Translational Genomics Unit at MCRI.

This analysis will focus on data generated from the left ventricle from three groups: fetal (19-20 weeks, n=3; 27,136 nuclei), non-diseased (ND; 4-14 years, n=3; 16,445 nuclei) and early-onset DCM samples (5-10 years, n=4; 31,593 nuclei). 

There are several aspects of the analysis:
* Define the cell types present at each group.
* Analyse differences in cell type composition across these groups
* Identify transcriptional pathways perturbed in children with dilated cardiomyopathy
* Define fetal gene program in dilated cardiomyopathy 

Also, we have developed an interactive website that permits interrogation of our data set and, hence, increases the data setâ€™s accessibility and utility. To check our website, please visit [www.HeartExplorer.org](http://www.HeartExplorer.org/).

A [workflowr][] project.

[workflowr]: https://github.com/workflowr/workflowr
