---
title: "Analysis of DCM samples"
author: "Belinda Phipson"
date: "6/21/2019"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


# Introduction

I will cluster all the DCM samples from batch 1 and batch 2 using the integration technique from the Seurat package.

# Load libraries and functions

```{r}
library(edgeR)
library(RColorBrewer)
library(org.Hs.eg.db)
library(limma)
library(Seurat)
library(cowplot)
library(DelayedArray)
library(scran)
library(NMF)
library(workflowr)
library(ggplot2)
library(clustree)
library(dplyr)
```

```{r}
source("code/normCounts.R")
source("code/findModes.R")
source("code/ggplotColors.R")
```

# Read in the DCM data

```{r}
targets <- read.delim("data/targets.txt",header=TRUE, stringsAsFactors = FALSE)
targets$FileName2 <- paste(targets$FileName,"/",sep="")
targets$Group_ID2 <- gsub("LV_","",targets$Group_ID)
group <- c("fetal_1","fetal_2","fetal_3",
           "non-diseased_1","non-diseased_2","non-diseased_3",
           "diseased_1","diseased_2",
           "diseased_3","diseased_4")
m <- match(group, targets$Group_ID2)
targets <- targets[m,]
```

```{r readData}
d1 <- Read10X(data.dir = targets$FileName2[targets$Group_ID2=="diseased_1"])
colnames(d1) <- paste(colnames(d1),"d1",sep="_")
d2 <- Read10X(data.dir = targets$FileName2[targets$Group_ID2=="diseased_2"])
colnames(d2) <- paste(colnames(d2),"d2",sep="_")
d3 <- Read10X(data.dir = targets$FileName2[targets$Group_ID2=="diseased_3"])
colnames(d3) <- paste(colnames(d3),"d3",sep="_")
d4 <- Read10X(data.dir = targets$FileName2[targets$Group_ID2=="diseased_4"])
colnames(d4) <- paste(colnames(d4),"d4",sep="_")

# Combine 4 samples into one big data matrix
alld <- cbind(d1,d2,d3,d4)
```

# Gene filtering

## Get gene annotation

I'm using gene annotation information from the `org.Hs.eg.db` package.

```{r,warning=FALSE}
columns(org.Hs.eg.db)
ann <- AnnotationDbi:::select(org.Hs.eg.db,keys=rownames(alld),columns=c("SYMBOL","ENTREZID","ENSEMBL","GENENAME","CHR"),keytype = "SYMBOL")
m <- match(rownames(alld),ann$SYMBOL)
ann <- ann[m,]
table(ann$SYMBOL==rownames(alld))
```

```{r}
mito <- grep("mitochondrial",ann$GENENAME)
length(mito)
ribo <- grep("ribosomal",ann$GENENAME)
length(ribo)
missingEZID <- which(is.na(ann$ENTREZID))
length(missingEZID)
```

## Remove mitochondrial and ribosomal genes and genes with no ENTREZID

These genes are not informative for downstream analysis.

```{r}
chuck <- unique(c(mito,ribo,missingEZID))
length(chuck)
alld.keep <- alld[-chuck,]
ann.keep <- ann[-chuck,]
table(ann.keep$SYMBOL==rownames(alld.keep))
```

## Remove very lowly expressed genes

Removing very lowly expressed genes helps to reduce the noise in the data. Here I am choosing to keep genes with at least 1 count in at least 20 cells. This means that a cluster made up of at least 20 cells can potentially be detected (minimum cluster size = 20 cells).

```{r}
numzero.genes <- rowSums(alld.keep==0)

#avg.exp <- rowMeans(cpm.DGEList(y.kid,log=TRUE))

#plot(avg.exp,numzero.genes,xlab="Average log-normalised-counts",ylab="Number zeroes per gene")

table(numzero.genes > (ncol(alld.keep)-20))

keep.genes <- numzero.genes < (ncol(alld.keep)-20)
table(keep.genes)

alld.keep <- alld.keep[keep.genes,]
dim(alld.keep)

ann.keep <- ann.keep[keep.genes,]
```

The total size of the DCM dataset is `r ncol(alld.keep)` cells and `r nrow(alld.keep)` genes.

# Remove sex chromosome genes

I will remove the sex chromosome genes before clustering so that the sex doesn't play a role in determining the clusters.

```{r}
sexchr <- ann.keep$CHR %in% c("X","Y")

alld.nosex <- alld.keep[!sexchr,]
dim(alld.nosex)

ann.nosex <- ann.keep[!sexchr,]
```

# Save data objects

```{r}
#save(ann,ann.keep,ann.nosex,alld,alld.keep,alld.nosex,file="./output/RDataObjects/dcmObjs.Rdata")
```

# Create Seurat objects

Here I am following the Seurat vignette on performing clustering using their integration method to deal with batch effects.

```{r}
biorep <- factor(rep(c("d1","d2","d3","d4"),c(ncol(d1),ncol(d2),ncol(d3),ncol(d4))))
names(biorep) <- colnames(alld.keep)
sex <- factor(rep(c("m","f","f","f"),c(ncol(d1),ncol(d2),ncol(d3),ncol(d4))))
names(sex) <- colnames(alld.keep)
age <- rep(c(5,10.83,9,10.83),c(ncol(d1),ncol(d2),ncol(d3),ncol(d4)))
names(age) <- colnames(alld.keep)
batch <- rep(c("B2","B2","B1","B2"),c(ncol(d1),ncol(d2),ncol(d3),ncol(d4)))
names(batch) <- colnames(alld.keep)

dcm <- CreateSeuratObject(counts = alld.nosex, project = "dcm")
dcm <- AddMetaData(object=dcm, metadata = biorep, col.name="biorep")
dcm <- AddMetaData(object=dcm, metadata = sex, col.name="sex")
dcm <- AddMetaData(object=dcm, metadata = age, col.name="age")
dcm <- AddMetaData(object=dcm, metadata = batch, col.name="batch")
```

```{r}
dcm.list <- SplitObject(dcm, split.by = "biorep")
```

# Try new normalisation method: SCTransform

This new method replaces the `NormalizeData`, `FindVariableFeatures` and `ScaleData` functions. It performs regularised negative binomial regression with the total sequencing depth per cell as the covariate (i.e. library size), as well as any other user supplied covariates. The Pearson residuals are then used in downstream analysis.


```{r sctnorm,warning=FALSE}
# This is a bit slow
for(i in 1:length(dcm.list)) {
    dcm.list[[i]] <- SCTransform(dcm.list[[i]], verbose = FALSE)
#    dcm.list[[i]] <- GetResidual(dcm.list[[i]])
}
```

# Perform the usual normalisation

```{r}
#for (i in 1:length(dcm.list)) {
#    dcm.list[[i]] <- NormalizeData(dcm.list[[i]], verbose = FALSE)
#    dcm.list[[i]] <- FindVariableFeatures(dcm.list[[i]], selection.method #= "vst", 
#                                            nfeatures = 2000, verbose = FALSE)
#}
```


# Perform integration

There are two steps:

* Find integration anchors
* Perform integration which should batch-correct the data

The default number of dimensions is 30.
Should increase the number of integration anchors to 3000.

```{r intanchors}
dcm.anchors <- FindIntegrationAnchors(object.list = dcm.list, dims = 1:30, anchor.features = 3000)
```

```{r integrate}
dcm.integrated <- IntegrateData(anchorset = dcm.anchors, dims = 1:30)
```

# Perform clustering

```{r}
DefaultAssay(object = dcm.integrated) <- "integrated"
```

## Perform scaling and PCA

```{r pca}
dcm.integrated <- ScaleData(dcm.integrated, verbose = FALSE)
dcm.integrated <- RunPCA(dcm.integrated, npcs = 50, verbose = FALSE)
ElbowPlot(dcm.integrated,ndims=50)
```

```{r,fig.height=10, fig.width=12}
VizDimLoadings(dcm.integrated, dims = 1:4, reduction = "pca")
```

```{r}
DimPlot(dcm.integrated, reduction = "pca",group.by="biorep")
```

```{r}
DimPlot(dcm.integrated, reduction = "pca",group.by="sex")
```

```{r}
DimPlot(dcm.integrated, reduction = "pca",group.by="batch")
```


```{r,fig.height=10, fig.width=12}
DimHeatmap(dcm.integrated, dims = 1:15, cells = 500, balanced = TRUE)
```

```{r}
DimHeatmap(dcm.integrated, dims = 16:30, cells = 500, balanced = TRUE)
```


## Perform nearest neighbours clustering

```{r clust}
dcm.integrated <- FindNeighbors(dcm.integrated, dims = 1:20)
dcm.integrated <- FindClusters(dcm.integrated, resolution = 0.3)
table(Idents(dcm.integrated))
```

```{r}
barplot(table(Idents(dcm.integrated)),ylab="Number of cells",xlab="Clusters")
title("Number of cells in each cluster")
```

## Visualisation with tSNE

```{r tsne}
set.seed(10)
dcm.integrated <- RunTSNE(dcm.integrated, reduction = "pca", dims = 1:20)
```

```{r}
#pdf(file="./output/Figures/tsne-dcmALL.pdf",width=10,height=8,onefile = FALSE)
DimPlot(dcm.integrated, reduction = "tsne",label=TRUE,label.size = 6,pt.size = 0.5)+NoLegend()
#dev.off()
```

## Visualisation with UMAP

```{r umap}
set.seed(10)
dcm.integrated <- RunUMAP(dcm.integrated, reduction = "pca", dims = 1:20)
```

```{r}
#pdf(file="./output/Figures/umap-dcmALL.pdf",width=10,height=8,onefile = FALSE)
DimPlot(dcm.integrated, reduction = "umap",label=TRUE,label.size = 6,pt.size = 0.5)+NoLegend()
#dev.off()
```

```{r,fig.width=12,fig.height=5}
DimPlot(dcm.integrated, reduction = "umap", split.by = "biorep",label=TRUE,label.size = 5)+NoLegend()
```

```{r,fig.width=10,fig.height=6}
DimPlot(dcm.integrated, reduction = "umap", split.by = "sex",label=TRUE,label.size = 5)+NoLegend()
```

```{r, fig.width=12,fig.height=9}
DefaultAssay(dcm.integrated) <- "SCT"
FeaturePlot(dcm.integrated, reduction = "umap", features = c("TNNT2", "MYH6", "WT1", "PECAM1", "TBX3", "PDGFRA"), pt.size = 0.2, label = T, ncol = 3)
```


# Visualisation with clustree

```{r}
DefaultAssay(dcm.integrated) <- "integrated"
clusres <- c(0.1,0.2,0.3,0.4,0.5,0.6,0.7,0.8,0.9,1,1.1,1.2)
```
```{r clustree}
for(i in 1:length(clusres)){
  dcm.integrated <- FindClusters(dcm.integrated, 
                                   resolution = clusres[i])
}
```

```{r}
pct.male <- function(x) {mean(x=="m")}
pct.female <- function(x) {mean(x=="f")}
biorep1 <- function(x) {mean(x=="d1")}
biorep2 <- function(x) {mean(x=="d2")}
biorep3 <- function(x) {mean(x=="d3")}
biorep3 <- function(x) {mean(x=="d4")}
```

```{r,fig.width=10,fig.height=10}
clustree(dcm.integrated, prefix = "integrated_snn_res.")
```

```{r,fig.width=10,fig.height=10}
clustree(dcm.integrated, prefix = "integrated_snn_res.",
         node_colour = "biorep", node_colour_aggr = "biorep1",assay="RNA")
```

```{r,fig.width=10,fig.height=10}
clustree(dcm.integrated, prefix = "integrated_snn_res.",
         node_colour = "biorep", node_colour_aggr = "biorep2",assay="RNA")
```

```{r,fig.width=10,fig.height=10}
clustree(dcm.integrated, prefix = "integrated_snn_res.",
         node_colour = "biorep", node_colour_aggr = "biorep3",assay="RNA")
```


```{r,fig.width=10,fig.height=10}
clustree(dcm.integrated, prefix = "integrated_snn_res.",
         node_colour = "sex", node_colour_aggr = "pct.female",assay="RNA")
```

```{r,fig.width=10,fig.height=10}
clustree(dcm.integrated, prefix = "integrated_snn_res.",
         node_colour = "sex", node_colour_aggr = "pct.male",assay="RNA")
```


```{r,fig.width=10,fig.height=10}
clustree(dcm.integrated, prefix = "integrated_snn_res.",
         node_colour = "TNNT2", node_colour_aggr = "median",
         assay="RNA")
```

# Save Seurat object


```{r}
DefaultAssay(dcm.integrated) <- "RNA"
Idents(dcm.integrated) <- dcm.integrated$integrated_snn_res.0.3
```

```{r}
#saveRDS(dcm.integrated,file="./output/RDataObjects/dcm-int.Rds")
```


```{r,fig.width=12,fig.height=4}
par(mfrow=c(1,1))
par(mar=c(4,4,2,2))
tab <- table(Idents(dcm.integrated),dcm@meta.data$biorep)
barplot(t(tab/rowSums(tab)),beside=TRUE,col=ggplotColors(4),legend=TRUE)
```

```{r markers}
# Limma-trend for DE

y.dcm <- DGEList(alld.keep)

logcounts <- normCounts(y.dcm,log=TRUE,prior.count=0.5)

maxclust <- length(levels(Idents(dcm.integrated)))-1

grp <- paste("c",Idents(dcm.integrated),sep = "")
grp <- factor(grp,levels = paste("c",0:maxclust,sep=""))

design <- model.matrix(~0+grp)
colnames(design) <- levels(grp)

mycont <- matrix(NA,ncol=length(levels(grp)),nrow=length(levels(grp)))
rownames(mycont)<-colnames(mycont)<-levels(grp)
diag(mycont)<-1
mycont[upper.tri(mycont)]<- -1/(length(levels(factor(grp)))-1)
mycont[lower.tri(mycont)]<- -1/(length(levels(factor(grp)))-1)

fit <- lmFit(logcounts,design)
fit.cont <- contrasts.fit(fit,contrasts=mycont)
fit.cont <- eBayes(fit.cont,trend=TRUE,robust=TRUE)

fit.cont$genes <- ann.keep

summary(decideTests(fit.cont))
```

```{r}
treat <- treat(fit.cont,lfc=0.5)

dt <- decideTests(treat)

summary(dt)
```


```{r,fig.width=10,fig.height=10}
par(mfrow=c(3,3))
for(i in 1:ncol(mycont)){
  plotMD(treat,coef=i,status = dt[,i],hl.cex=0.5)
  abline(h=0,col=colours()[c(226)])
  lines(lowess(treat$Amean,treat$coefficients[,i]),lwd=1.5,col=4)
}
```

## Write out marker genes for each cluster

```{rm}
#write.csv(dcmmarkers,file="./output/Alldcm-clustermarkers.csv")
contnames <- colnames(mycont)

for(i in 1:length(contnames)){
  topsig <- topTreat(treat,coef=i,n=Inf,p.value=0.05)
  write.csv(topsig[topsig$logFC>0,],file=paste("./output/Up-Cluster-",contnames[i],".csv",sep=""))
  write.csv(topGO(goana(de=topsig$ENTREZID[topsig$logFC>0],universe=treat$genes$ENTREZID,species="Hs"),number=50),
            file=paste("./output/GO-Cluster-",contnames[i],".csv",sep=""))
}

```

# Heatmap of pre-defined heart cell type markers

```{r}
hm <- read.delim("./data/heart-markers-long.txt",stringsAsFactors = FALSE)
hgene <- toupper(hm$Gene)
hgene <- unique(hgene)

m <- match(hgene,rownames(logcounts))
m <- m[!is.na(m)]

sam <- factor(dcm.integrated$biorep)
newgrp <- paste(grp,sam,sep=".")
newgrp <- factor(newgrp,levels=paste(rep(levels(grp),each=4),levels(sam),sep="."))
o <-order(newgrp)

annot <- data.frame(CellType=grp,Sample=sam,NewGroup=newgrp)

mycelltypes <- hm$Celltype[match(rownames(logcounts)[m],toupper(hm$Gene))]
mycelltypes <- factor(mycelltypes)


myColors <- list(Clust=NA,Sample=NA,Celltypes=NA)
myColors$Clust<-sample(ggplotColors(length(levels(grp))),length(levels(grp)))
names(myColors$Clust)<-levels(grp)
myColors$Sample <- brewer.pal(4, "Set1")
names(myColors$Sample) <- levels(sam)
myColors$Celltypes <- ggplotColors(length(levels(mycelltypes)))
names(myColors$Celltypes) <- levels(mycelltypes) 

mygenes <- rownames(logcounts)[m]
mygenelab <- paste(mygenes,mycelltypes,sep="_")
```

```{r heatmapHMarkers}
par(mfrow=c(1,1))
#pdf(file="./output/Figures/dcm-heatmap-hmarkers-res03.pdf",width=20,height=15,onefile = FALSE)
aheatmap(logcounts[m,o],Rowv=NA,Colv=NA,labRow=mygenelab,labCol=NA,
         annCol=list(Clust=grp[o],Sample=sam[o]),
#         annRow = list(Celltypes=mycelltypes),
         annColors = myColors, 
         fontsize=10,color="-RdYlBu")
#dev.off()
```

# Summarise expression across cells 
```{r}
sumexpr <- matrix(NA,nrow=nrow(logcounts),ncol=length(levels(newgrp)))
rownames(sumexpr) <- rownames(logcounts)
colnames(sumexpr) <- levels(newgrp)

for(i in 1:nrow(sumexpr)){
  sumexpr[i,] <- tapply(logcounts[i,],newgrp,median)
}

```


```{r}
par(mfrow=c(1,1))
clust <- rep(levels(grp),each=4)
samps <- rep(levels(sam),length(levels(grp)))
#pdf(file="./output/Figures/dcm-heatmap-hmarkers-summarised-res03.pdf",width=20,height=15,onefile = FALSE)
aheatmap(sumexpr[m,],Rowv = NA,Colv = NA, labRow = mygenelab,
         annCol=list(Clust=clust,Sample=samps),
 #        annRow=list(Celltypes=mycelltypes),
         annColors=myColors,
         fontsize=10,color="-RdYlBu")
#dev.off()
```

# Try annotate clusters based on fetal cell types (Compare DCM to fetal)

```{r}
# This loads a list object called sig.genes.gst
load(file="/group/card2/Neda/MCRI_LAB/scRNAseq-ES/Data/gstlist-fetal.Rdata")

expr.score <- matrix(NA,nrow=length(sig.genes.gst),ncol=length(levels(newgrp)))
colnames(expr.score) <- levels(newgrp)
rownames(expr.score) <- names(sig.genes.gst)

specialm <- lapply(sig.genes.gst,function(x) match(x,rownames(sumexpr))[!is.na(match(x,rownames(sumexpr)))])

for(i in 1:nrow(expr.score)){
  expr.score[i,] <- colMeans(sumexpr[specialm[[i]],])  
}

#pdf(file="./output/Figures/heatmap-match-fetal-dcm-means-res03.pdf",width=20,height=15,onefile = FALSE)
aheatmap(expr.score,
         Rowv = NA,Colv = NA, 
         labRow = rownames(expr.score),
         annCol=list(Clust=clust,Sample=samps),
         annColors=myColors,
         fontsize=10,color="-RdYlBu",
         scale="none")
#dev.off()
```

# Compare DCM to ND

```{r}
# This loads a list object called non-diseased.sig.genes.gst
#non-diseased samples are the young samples already published in Sim et al., 2021
load(file="/group/card2/Neda/MCRI_LAB/scRNAseq-ES/Data/gstlist-ND.Rdata")

expr.score <- matrix(NA,nrow=length(non.diseased.sig.genes.gst),ncol=length(levels(newgrp)))
colnames(expr.score) <- levels(newgrp)
rownames(expr.score) <- names(non.diseased.sig.genes.gst)

specialm <- lapply(non.diseased.sig.genes.gst,function(x) match(x,rownames(sumexpr))[!is.na(match(x,rownames(sumexpr)))])

for(i in 1:nrow(expr.score)){
  expr.score[i,] <- colMedians(sumexpr[specialm[[i]],])  
}

#pdf(file="./output/Figures/heatmap-match-nd-dcm-medians.pdf",width=20,height=15,onefile = FALSE)
aheatmap(expr.score,
         Rowv = NA,Colv = NA, 
         labRow = rownames(expr.score),
         annCol=list(Clust=clust,Sample=samps),
         annColors=myColors,
         fontsize=10,color="-RdYlBu",
         scale="none")
#dev.off()
```

# Create list of marker genes for GST purposes

```{r}
dcm.sig.genes.gst <- vector("list", length(levels(grp)))
names(dcm.sig.genes.gst) <- levels(grp)
for(i in 1:length(dcm.sig.genes.gst)){
  top <- topTreat(treat,coef=i,n=Inf,p.value=0.05)
  dcm.sig.genes.gst[[i]] <- rownames(top)[top$logFC>0]
} 
#save(dcm.sig.genes.gst,file="./data/gstlist-dcm.Rdata")
```



